// three-stdlib/dist/index.js
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const MD2CharacterComplex = require("./misc/MD2CharacterComplex.js");
const ConvexObjectBreaker = require("./misc/ConvexObjectBreaker.js");
const MorphBlendMesh = require("./misc/MorphBlendMesh.js");
const GPUComputationRenderer = require("./misc/GPUComputationRenderer.js");
const Gyroscope = require("./misc/Gyroscope.js");
const MorphAnimMesh = require("./misc/MorphAnimMesh.js");
const RollerCoaster = require("./misc/RollerCoaster.js");
const Timer = require("./misc/Timer.js");
const WebGL = require("./misc/WebGL.js");
const MD2Character = require("./misc/MD2Character.js");
const VolumeSlice = require("./misc/VolumeSlice.js");
const TubePainter = require("./misc/TubePainter.js");
const Volume = require("./misc/Volume.js");
const ProgressiveLightmap = require("./misc/ProgressiveLightmap.js");
const CSS2DRenderer = require("./renderers/CSS2DRenderer.js");
const CSS3DRenderer = require("./renderers/CSS3DRenderer.js");
const Projector = require("./renderers/Projector.js");
const SVGRenderer = require("./renderers/SVGRenderer.js");
const FlakesTexture = require("./textures/FlakesTexture.js");
const CurveModifier = require("./modifiers/CurveModifier.js");
const SimplifyModifier = require("./modifiers/SimplifyModifier.js");
const EdgeSplitModifier = require("./modifiers/EdgeSplitModifier.js");
const TessellateModifier = require("./modifiers/TessellateModifier.js");
const GLTFExporter = require("./exporters/GLTFExporter.js");
const USDZExporter = require("./exporters/USDZExporter.js");
const PLYExporter = require("./exporters/PLYExporter.js");
const DRACOExporter = require("./exporters/DRACOExporter.js");
const ColladaExporter = require("./exporters/ColladaExporter.js");
const MMDExporter = require("./exporters/MMDExporter.js");
const STLExporter = require("./exporters/STLExporter.js");
const OBJExporter = require("./exporters/OBJExporter.js");
const RoomEnvironment = require("./environments/RoomEnvironment.js");
const AnimationClipCreator = require("./animation/AnimationClipCreator.js");
const CCDIKSolver = require("./animation/CCDIKSolver.js");
const MMDPhysics = require("./animation/MMDPhysics.js");
const MMDAnimationHelper = require("./animation/MMDAnimationHelper.js");
const Reflector = require("./objects/Reflector.js");
const Refractor = require("./objects/Refractor.js");
const ShadowMesh = require("./objects/ShadowMesh.js");
const Lensflare = require("./objects/Lensflare.js");
const Water = require("./objects/Water.js");
const MarchingCubes = require("./objects/MarchingCubes.js");
const LightningStorm = require("./objects/LightningStorm.js");
const ReflectorRTT = require("./objects/ReflectorRTT.js");
const ReflectorForSSRPass = require("./objects/ReflectorForSSRPass.js");
const Sky = require("./objects/Sky.js");
const Water2 = require("./objects/Water2.js");
const GroundProjectedEnv = require("./objects/GroundProjectedEnv.js");
const SceneUtils = require("./utils/SceneUtils.js");
const UVsDebug = require("./utils/UVsDebug.js");
const GeometryUtils = require("./utils/GeometryUtils.js");
const RoughnessMipmapper = require("./utils/RoughnessMipmapper.js");
const SkeletonUtils = require("./utils/SkeletonUtils.js");
const ShadowMapViewer = require("./utils/ShadowMapViewer.js");
const BufferGeometryUtils = require("./utils/BufferGeometryUtils.js");
const GeometryCompressionUtils = require("./utils/GeometryCompressionUtils.js");
const CinematicCamera = require("./cameras/CinematicCamera.js");
const ConvexHull = require("./math/ConvexHull.js");
const MeshSurfaceSampler = require("./math/MeshSurfaceSampler.js");
const SimplexNoise = require("./math/SimplexNoise.js");
const OBB = require("./math/OBB.js");
const Capsule = require("./math/Capsule.js");
const ColorConverter = require("./math/ColorConverter.js");
const ImprovedNoise = require("./math/ImprovedNoise.js");
const Octree = require("./math/Octree.js");
const Lut = require("./math/Lut.js");
const CameraControls = require("./controls/experimental/CameraControls.js");
const FirstPersonControls = require("./controls/FirstPersonControls.js");
const TransformControls = require("./controls/TransformControls.js");
const DragControls = require("./controls/DragControls.js");
const PointerLockControls = require("./controls/PointerLockControls.js");
const DeviceOrientationControls = require("./controls/DeviceOrientationControls.js");
const TrackballControls = require("./controls/TrackballControls.js");
const OrbitControls = require("./controls/OrbitControls.js");
const ArcballControls = require("./controls/ArcballControls.js");
const FlyControls = require("./controls/FlyControls.js");
const LUTPass = require("./postprocessing/LUTPass.js");
const ClearPass = require("./postprocessing/ClearPass.js");
const GlitchPass = require("./postprocessing/GlitchPass.js");
const HalftonePass = require("./postprocessing/HalftonePass.js");
const SMAAPass = require("./postprocessing/SMAAPass.js");
const FilmPass = require("./postprocessing/FilmPass.js");
const OutlinePass = require("./postprocessing/OutlinePass.js");
const SSAOPass = require("./postprocessing/SSAOPass.js");
const SavePass = require("./postprocessing/SavePass.js");
const BokehPass = require("./postprocessing/BokehPass.js");
const Pass = require("./postprocessing/Pass.js");
const TexturePass = require("./postprocessing/TexturePass.js");
const AdaptiveToneMappingPass = require("./postprocessing/AdaptiveToneMappingPass.js");
const UnrealBloomPass = require("./postprocessing/UnrealBloomPass.js");
const CubeTexturePass = require("./postprocessing/CubeTexturePass.js");
const SAOPass = require("./postprocessing/SAOPass.js");
const AfterimagePass = require("./postprocessing/AfterimagePass.js");
const MaskPass = require("./postprocessing/MaskPass.js");
const EffectComposer = require("./postprocessing/EffectComposer.js");
const DotScreenPass = require("./postprocessing/DotScreenPass.js");
const SSRPass = require("./postprocessing/SSRPass.js");
const TAARenderPass = require("./postprocessing/TAARenderPass.js");
const ShaderPass = require("./postprocessing/ShaderPass.js");
const SSAARenderPass = require("./postprocessing/SSAARenderPass.js");
const RenderPass = require("./postprocessing/RenderPass.js");
const RenderPixelatedPass = require("./postprocessing/RenderPixelatedPass.js");
const BloomPass = require("./postprocessing/BloomPass.js");
const WaterPass = require("./postprocessing/WaterPass.js");
const ARButton = require("./webxr/ARButton.js");
const OculusHandModel = require("./webxr/OculusHandModel.js");
const OculusHandPointerModel = require("./webxr/OculusHandPointerModel.js");
const Text2D = require("./webxr/Text2D.js");
const VRButton = require("./webxr/VRButton.js");
const XRControllerModelFactory = require("./webxr/XRControllerModelFactory.js");
const XREstimatedLight = require("./webxr/XREstimatedLight.js");
const XRHandMeshModel = require("./webxr/XRHandMeshModel.js");
const XRHandModelFactory = require("./webxr/XRHandModelFactory.js");
const XRHandPrimitiveModel = require("./webxr/XRHandPrimitiveModel.js");
const ParametricGeometries = require("./geometries/ParametricGeometries.js");
const ParametricGeometry = require("./geometries/ParametricGeometry.js");
const ConvexGeometry = require("./geometries/ConvexGeometry.js");
const LightningStrike = require("./geometries/LightningStrike.js");
const RoundedBoxGeometry = require("./geometries/RoundedBoxGeometry.js");
const BoxLineGeometry = require("./geometries/BoxLineGeometry.js");
const DecalGeometry = require("./geometries/DecalGeometry.js");
const TeapotGeometry = require("./geometries/TeapotGeometry.js");
const TextGeometry = require("./geometries/TextGeometry.js");
const CSM = require("./csm/CSM.js");
const CSMFrustum = require("./csm/CSMFrustum.js");
const CSMHelper = require("./csm/CSMHelper.js");
const CSMShader = require("./csm/CSMShader.js");
require("./shaders/types.js");
const ACESFilmicToneMappingShader = require("./shaders/ACESFilmicToneMappingShader.js");
const AfterimageShader = require("./shaders/AfterimageShader.js");
const BasicShader = require("./shaders/BasicShader.js");
const BleachBypassShader = require("./shaders/BleachBypassShader.js");
const BlendShader = require("./shaders/BlendShader.js");
const BokehShader = require("./shaders/BokehShader.js");
const BokehShader2 = require("./shaders/BokehShader2.js");
const BrightnessContrastShader = require("./shaders/BrightnessContrastShader.js");
const ColorCorrectionShader = require("./shaders/ColorCorrectionShader.js");
const ColorifyShader = require("./shaders/ColorifyShader.js");
const ConvolutionShader = require("./shaders/ConvolutionShader.js");
const CopyShader = require("./shaders/CopyShader.js");
const DOFMipMapShader = require("./shaders/DOFMipMapShader.js");
const DepthLimitedBlurShader = require("./shaders/DepthLimitedBlurShader.js");
const DigitalGlitch = require("./shaders/DigitalGlitch.js");
const DotScreenShader = require("./shaders/DotScreenShader.js");
const FXAAShader = require("./shaders/FXAAShader.js");
const FilmShader = require("./shaders/FilmShader.js");
const FocusShader = require("./shaders/FocusShader.js");
const FreiChenShader = require("./shaders/FreiChenShader.js");
const FresnelShader = require("./shaders/FresnelShader.js");
const GammaCorrectionShader = require("./shaders/GammaCorrectionShader.js");
const GodRaysShader = require("./shaders/GodRaysShader.js");
const HalftoneShader = require("./shaders/HalftoneShader.js");
const HorizontalBlurShader = require("./shaders/HorizontalBlurShader.js");
const HorizontalTiltShiftShader = require("./shaders/HorizontalTiltShiftShader.js");
const HueSaturationShader = require("./shaders/HueSaturationShader.js");
const KaleidoShader = require("./shaders/KaleidoShader.js");
const LuminosityHighPassShader = require("./shaders/LuminosityHighPassShader.js");
const LuminosityShader = require("./shaders/LuminosityShader.js");
const MirrorShader = require("./shaders/MirrorShader.js");
const NormalMapShader = require("./shaders/NormalMapShader.js");
const ParallaxShader = require("./shaders/ParallaxShader.js");
const PixelShader = require("./shaders/PixelShader.js");
const RGBShiftShader = require("./shaders/RGBShiftShader.js");
const SAOShader = require("./shaders/SAOShader.js");
const SMAAShader = require("./shaders/SMAAShader.js");
const SSAOShader = require("./shaders/SSAOShader.js");
const SSRShader = require("./shaders/SSRShader.js");
const SepiaShader = require("./shaders/SepiaShader.js");
const SobelOperatorShader = require("./shaders/SobelOperatorShader.js");
const SubsurfaceScatteringShader = require("./shaders/SubsurfaceScatteringShader.js");
const TechnicolorShader = require("./shaders/TechnicolorShader.js");
const ToneMapShader = require("./shaders/ToneMapShader.js");
const ToonShader = require("./shaders/ToonShader.js");
const TriangleBlurShader = require("./shaders/TriangleBlurShader.js");
const UnpackDepthRGBAShader = require("./shaders/UnpackDepthRGBAShader.js");
const VerticalBlurShader = require("./shaders/VerticalBlurShader.js");
const VerticalTiltShiftShader = require("./shaders/VerticalTiltShiftShader.js");
const VignetteShader = require("./shaders/VignetteShader.js");
const VolumeShader = require("./shaders/VolumeShader.js");
const WaterRefractionShader = require("./shaders/WaterRefractionShader.js");
const HTMLMesh = require("./interactive/HTMLMesh.js");
const InteractiveGroup = require("./interactive/InteractiveGroup.js");
const SelectionHelper = require("./interactive/SelectionHelper.js");
const SelectionBox = require("./interactive/SelectionBox.js");
const AmmoPhysics = require("./physics/AmmoPhysics.js");
const ParallaxBarrierEffect = require("./effects/ParallaxBarrierEffect.js");
const PeppersGhostEffect = require("./effects/PeppersGhostEffect.js");
const OutlineEffect = require("./effects/OutlineEffect.js");
const AnaglyphEffect = require("./effects/AnaglyphEffect.js");
const AsciiEffect = require("./effects/AsciiEffect.js");
const StereoEffect = require("./effects/StereoEffect.js");
const FBXLoader = require("./loaders/FBXLoader.js");
const FontLoader = require("./loaders/FontLoader.js");
const TGALoader = require("./loaders/TGALoader.js");
const LUTCubeLoader = require("./loaders/LUTCubeLoader.js");
const NRRDLoader = require("./loaders/NRRDLoader.js");
const STLLoader = require("./loaders/STLLoader.js");
const MTLLoader = require("./loaders/MTLLoader.js");
const XLoader = require("./loaders/XLoader.js");
const BVHLoader = require("./loaders/BVHLoader.js");
const KMZLoader = require("./loaders/KMZLoader.js");
const VRMLoader = require("./loaders/VRMLoader.js");
const VRMLLoader = require("./loaders/VRMLLoader.js");
const KTX2Loader = require("./loaders/KTX2Loader.js");
const LottieLoader = require("./loaders/LottieLoader.js");
const TTFLoader = require("./loaders/TTFLoader.js");
const RGBELoader = require("./loaders/RGBELoader.js");
const AssimpLoader = require("./loaders/AssimpLoader.js");
const ColladaLoader = require("./loaders/ColladaLoader.js");
const MDDLoader = require("./loaders/MDDLoader.js");
const EXRLoader = require("./loaders/EXRLoader.js");
const _3MFLoader = require("./loaders/3MFLoader.js");
const XYZLoader = require("./loaders/XYZLoader.js");
const VTKLoader = require("./loaders/VTKLoader.js");
const LUT3dlLoader = require("./loaders/LUT3dlLoader.js");
const DDSLoader = require("./loaders/DDSLoader.js");
const PVRLoader = require("./loaders/PVRLoader.js");
const GCodeLoader = require("./loaders/GCodeLoader.js");
const BasisTextureLoader = require("./loaders/BasisTextureLoader.js");
const TDSLoader = require("./loaders/TDSLoader.js");
const LDrawLoader = require("./loaders/LDrawLoader.js");
const GLTFLoader = require("./loaders/GLTFLoader.js");
const SVGLoader = require("./loaders/SVGLoader.js");
const _3DMLoader = require("./loaders/3DMLoader.js");
const OBJLoader = require("./loaders/OBJLoader.js");
const AMFLoader = require("./loaders/AMFLoader.js");
const MMDLoader = require("./loaders/MMDLoader.js");
const MD2Loader = require("./loaders/MD2Loader.js");
const KTXLoader = require("./loaders/KTXLoader.js");
const TiltLoader = require("./loaders/TiltLoader.js");
const DRACOLoader = require("./loaders/DRACOLoader.js");
const HDRCubeTextureLoader = require("./loaders/HDRCubeTextureLoader.js");
const PDBLoader = require("./loaders/PDBLoader.js");
const PRWMLoader = require("./loaders/PRWMLoader.js");
const RGBMLoader = require("./loaders/RGBMLoader.js");
const VOXLoader = require("./loaders/VOXLoader.js");
const PCDLoader = require("./loaders/PCDLoader.js");
const LWOLoader = require("./loaders/LWOLoader.js");
const PLYLoader = require("./loaders/PLYLoader.js");
const LineSegmentsGeometry = require("./lines/LineSegmentsGeometry.js");
const LineGeometry = require("./lines/LineGeometry.js");
const Wireframe = require("./lines/Wireframe.js");
const WireframeGeometry2 = require("./lines/WireframeGeometry2.js");
const Line2 = require("./lines/Line2.js");
const LineMaterial = require("./lines/LineMaterial.js");
const LineSegments2 = require("./lines/LineSegments2.js");
const LightProbeHelper = require("./helpers/LightProbeHelper.js");
const VertexTangentsHelper = require("./helpers/VertexTangentsHelper.js");
const PositionalAudioHelper = require("./helpers/PositionalAudioHelper.js");
const VertexNormalsHelper = require("./helpers/VertexNormalsHelper.js");
const RectAreaLightHelper = require("./helpers/RectAreaLightHelper.js");
const RectAreaLightUniformsLib = require("./lights/RectAreaLightUniformsLib.js");
const LightProbeGenerator = require("./lights/LightProbeGenerator.js");
const NURBSUtils = require("./curves/NURBSUtils.js");
const NURBSCurve = require("./curves/NURBSCurve.js");
const NURBSSurface = require("./curves/NURBSSurface.js");
const CurveExtras = require("./curves/CurveExtras.js");
const Geometry = require("./deprecated/Geometry.js");
const MeshoptDecoder = require("./libs/MeshoptDecoder.js");
const MotionControllers = require("./libs/MotionControllers.js");
exports.MD2CharacterComplex = MD2CharacterComplex.MD2CharacterComplex;
exports.ConvexObjectBreaker = ConvexObjectBreaker.ConvexObjectBreaker;
exports.MorphBlendMesh = MorphBlendMesh.MorphBlendMesh;
exports.GPUComputationRenderer = GPUComputationRenderer.GPUComputationRenderer;
exports.Gyroscope = Gyroscope.Gyroscope;
exports.MorphAnimMesh = MorphAnimMesh.MorphAnimMesh;
exports.RollerCoasterGeometry = RollerCoaster.RollerCoasterGeometry;
exports.RollerCoasterLiftersGeometry = RollerCoaster.RollerCoasterLiftersGeometry;
exports.RollerCoasterShadowGeometry = RollerCoaster.RollerCoasterShadowGeometry;
exports.SkyGeometry = RollerCoaster.SkyGeometry;
exports.TreesGeometry = RollerCoaster.TreesGeometry;
exports.Timer = Timer.Timer;
exports.getErrorMessage = WebGL.getErrorMessage;
exports.getWebGL2ErrorMessage = WebGL.getWebGL2ErrorMessage;
exports.getWebGLErrorMessage = WebGL.getWebGLErrorMessage;
exports.isWebGL2Available = WebGL.isWebGL2Available;
exports.isWebGLAvailable = WebGL.isWebGLAvailable;
exports.MD2Character = MD2Character.MD2Character;
exports.VolumeSlice = VolumeSlice.VolumeSlice;
exports.TubePainter = TubePainter.TubePainter;
exports.Volume = Volume.Volume;
exports.ProgressiveLightMap = ProgressiveLightmap.ProgressiveLightMap;
exports.CSS2DObject = CSS2DRenderer.CSS2DObject;
exports.CSS2DRenderer = CSS2DRenderer.CSS2DRenderer;
exports.CSS3DObject = CSS3DRenderer.CSS3DObject;
exports.CSS3DRenderer = CSS3DRenderer.CSS3DRenderer;
exports.CSS3DSprite = CSS3DRenderer.CSS3DSprite;
exports.Projector = Projector.Projector;
exports.RenderableFace = Projector.RenderableFace;
exports.RenderableLine = Projector.RenderableLine;
exports.RenderableObject = Projector.RenderableObject;
exports.RenderableSprite = Projector.RenderableSprite;
exports.RenderableVertex = Projector.RenderableVertex;
exports.SVGObject = SVGRenderer.SVGObject;
exports.SVGRenderer = SVGRenderer.SVGRenderer;
exports.FlakesTexture = FlakesTexture.FlakesTexture;
exports.Flow = CurveModifier.Flow;
exports.InstancedFlow = CurveModifier.InstancedFlow;
exports.getUniforms = CurveModifier.getUniforms;
exports.initSplineTexture = CurveModifier.initSplineTexture;
exports.modifyShader = CurveModifier.modifyShader;
exports.updateSplineTexture = CurveModifier.updateSplineTexture;
exports.SimplifyModifier = SimplifyModifier.SimplifyModifier;
exports.EdgeSplitModifier = EdgeSplitModifier.EdgeSplitModifier;
exports.TessellateModifier = TessellateModifier.TessellateModifier;
exports.GLTFExporter = GLTFExporter.GLTFExporter;
exports.USDZExporter = USDZExporter.USDZExporter;
exports.PLYExporter = PLYExporter.PLYExporter;
exports.DRACOExporter = DRACOExporter.DRACOExporter;
exports.ColladaExporter = ColladaExporter.ColladaExporter;
exports.MMDExporter = MMDExporter.MMDExporter;
exports.STLExporter = STLExporter.STLExporter;
exports.OBJExporter = OBJExporter.OBJExporter;
exports.RoomEnvironment = RoomEnvironment.RoomEnvironment;
exports.AnimationClipCreator = AnimationClipCreator.AnimationClipCreator;
exports.CCDIKHelper = CCDIKSolver.CCDIKHelper;
exports.CCDIKSolver = CCDIKSolver.CCDIKSolver;
exports.MMDPhysics = MMDPhysics.MMDPhysics;
exports.MMDAnimationHelper = MMDAnimationHelper.MMDAnimationHelper;
exports.Reflector = Reflector.Reflector;
exports.Refractor = Refractor.Refractor;
exports.ShadowMesh = ShadowMesh.ShadowMesh;
exports.Lensflare = Lensflare.Lensflare;
exports.LensflareElement = Lensflare.LensflareElement;
exports.Water = Water.Water;
exports.MarchingCubes = MarchingCubes.MarchingCubes;
exports.edgeTable = MarchingCubes.edgeTable;
exports.triTable = MarchingCubes.triTable;
exports.LightningStorm = LightningStorm.LightningStorm;
exports.ReflectorRTT = ReflectorRTT.ReflectorRTT;
exports.ReflectorForSSRPass = ReflectorForSSRPass.ReflectorForSSRPass;
exports.Sky = Sky.Sky;
exports.Water2 = Water2.Water2;
exports.GroundProjectedEnv = GroundProjectedEnv.GroundProjectedEnv;
exports.SceneUtils = SceneUtils.SceneUtils;
exports.UVsDebug = UVsDebug.UVsDebug;
exports.GeometryUtils = GeometryUtils.GeometryUtils;
exports.RoughnessMipmapper = RoughnessMipmapper.RoughnessMipmapper;
exports.SkeletonUtils = SkeletonUtils.SkeletonUtils;
exports.ShadowMapViewer = ShadowMapViewer.ShadowMapViewer;
exports.computeMorphedAttributes = BufferGeometryUtils.computeMorphedAttributes;
exports.estimateBytesUsed = BufferGeometryUtils.estimateBytesUsed;
exports.interleaveAttributes = BufferGeometryUtils.interleaveAttributes;
exports.mergeBufferAttributes = BufferGeometryUtils.mergeBufferAttributes;
exports.mergeBufferGeometries = BufferGeometryUtils.mergeBufferGeometries;
exports.mergeVertices = BufferGeometryUtils.mergeVertices;
exports.toCreasedNormals = BufferGeometryUtils.toCreasedNormals;
exports.toTrianglesDrawMode = BufferGeometryUtils.toTrianglesDrawMode;
exports.GeometryCompressionUtils = GeometryCompressionUtils.GeometryCompressionUtils;
exports.PackedPhongMaterial = GeometryCompressionUtils.PackedPhongMaterial;
exports.CinematicCamera = CinematicCamera.CinematicCamera;
exports.ConvexHull = ConvexHull.ConvexHull;
exports.Face = ConvexHull.Face;
exports.HalfEdge = ConvexHull.HalfEdge;
exports.VertexList = ConvexHull.VertexList;
exports.VertexNode = ConvexHull.VertexNode;
exports.MeshSurfaceSampler = MeshSurfaceSampler.MeshSurfaceSampler;
exports.SimplexNoise = SimplexNoise.SimplexNoise;
exports.OBB = OBB.OBB;
exports.Capsule = Capsule.Capsule;
exports.ColorConverter = ColorConverter.ColorConverter;
exports.ImprovedNoise = ImprovedNoise.ImprovedNoise;
exports.Octree = Octree.Octree;
exports.ColorMapKeywords = Lut.ColorMapKeywords;
exports.Lut = Lut.Lut;
exports.CameraControls = CameraControls.CameraControls;
exports.MapControlsExp = CameraControls.MapControlsExp;
exports.OrbitControlsExp = CameraControls.OrbitControlsExp;
exports.STATE = CameraControls.STATE;
exports.TrackballControlsExp = CameraControls.TrackballControlsExp;
exports.FirstPersonControls = FirstPersonControls.FirstPersonControls;
exports.TransformControls = TransformControls.TransformControls;
exports.TransformControlsGizmo = TransformControls.TransformControlsGizmo;
exports.TransformControlsPlane = TransformControls.TransformControlsPlane;
exports.DragControls = DragControls.DragControls;
exports.PointerLockControls = PointerLockControls.PointerLockControls;
exports.DeviceOrientationControls = DeviceOrientationControls.DeviceOrientationControls;
exports.TrackballControls = TrackballControls.TrackballControls;
exports.MapControls = OrbitControls.MapControls;
exports.OrbitControls = OrbitControls.OrbitControls;
exports.ArcballControls = ArcballControls.ArcballControls;
exports.FlyControls = FlyControls.FlyControls;
exports.LUTPass = LUTPass.LUTPass;
exports.ClearPass = ClearPass.ClearPass;
exports.GlitchPass = GlitchPass.GlitchPass;
exports.HalftonePass = HalftonePass.HalftonePass;
exports.SMAAPass = SMAAPass.SMAAPass;
exports.FilmPass = FilmPass.FilmPass;
exports.OutlinePass = OutlinePass.OutlinePass;
exports.SSAOPass = SSAOPass.SSAOPass;
exports.SavePass = SavePass.SavePass;
exports.BokehPass = BokehPass.BokehPass;
exports.FullScreenQuad = Pass.FullScreenQuad;
exports.Pass = Pass.Pass;
exports.TexturePass = TexturePass.TexturePass;
exports.AdaptiveToneMappingPass = AdaptiveToneMappingPass.AdaptiveToneMappingPass;
exports.UnrealBloomPass = UnrealBloomPass.UnrealBloomPass;
exports.CubeTexturePass = CubeTexturePass.CubeTexturePass;
exports.SAOPass = SAOPass.SAOPass;
exports.AfterimagePass = AfterimagePass.AfterimagePass;
exports.ClearMaskPass = MaskPass.ClearMaskPass;
exports.MaskPass = MaskPass.MaskPass;
exports.EffectComposer = EffectComposer.EffectComposer;
exports.DotScreenPass = DotScreenPass.DotScreenPass;
exports.SSRPass = SSRPass.SSRPass;
exports.TAARenderPass = TAARenderPass.TAARenderPass;
exports.ShaderPass = ShaderPass.ShaderPass;
exports.SSAARenderPass = SSAARenderPass.SSAARenderPass;
exports.RenderPass = RenderPass.RenderPass;
exports.RenderPixelatedPass = RenderPixelatedPass.RenderPixelatedPass;
exports.BloomPass = BloomPass.BloomPass;
exports.WaterPass = WaterPass.WaterPass;
exports.ARButton = ARButton.ARButton;
exports.OculusHandModel = OculusHandModel.OculusHandModel;
exports.OculusHandPointerModel = OculusHandPointerModel.OculusHandPointerModel;
exports.createText = Text2D.createText;
exports.VRButton = VRButton.VRButton;
exports.XRControllerModelFactory = XRControllerModelFactory.XRControllerModelFactory;
exports.XREstimatedLight = XREstimatedLight.XREstimatedLight;
exports.XRHandMeshModel = XRHandMeshModel.XRHandMeshModel;
exports.XRHandModelFactory = XRHandModelFactory.XRHandModelFactory;
exports.XRHandPrimitiveModel = XRHandPrimitiveModel.XRHandPrimitiveModel;
exports.ParametricGeometries = ParametricGeometries.ParametricGeometries;
exports.ParametricGeometry = ParametricGeometry.ParametricGeometry;
exports.ConvexGeometry = ConvexGeometry.ConvexGeometry;
exports.LightningStrike = LightningStrike.LightningStrike;
exports.RoundedBoxGeometry = RoundedBoxGeometry.RoundedBoxGeometry;
exports.BoxLineGeometry = BoxLineGeometry.BoxLineGeometry;
exports.DecalGeometry = DecalGeometry.DecalGeometry;
exports.DecalVertex = DecalGeometry.DecalVertex;
exports.TeapotGeometry = TeapotGeometry.TeapotGeometry;
exports.TextBufferGeometry = TextGeometry.TextGeometry;
exports.TextGeometry = TextGeometry.TextGeometry;
exports.CSM = CSM.CSM;
exports.CSMFrustum = CSMFrustum.CSMFrustum;
exports.CSMHelper = CSMHelper.CSMHelper;
exports.CSMShader = CSMShader.CSMShader;
exports.ACESFilmicToneMappingShader = ACESFilmicToneMappingShader.ACESFilmicToneMappingShader;
exports.AfterimageShader = AfterimageShader.AfterimageShader;
exports.BasicShader = BasicShader.BasicShader;
exports.BleachBypassShader = BleachBypassShader.BleachBypassShader;
exports.BlendShader = BlendShader.BlendShader;
exports.BokehShader = BokehShader.BokehShader;
exports.BokehDepthShader = BokehShader2.BokehDepthShader;
exports.BokehShader2 = BokehShader2.BokehShader2;
exports.BrightnessContrastShader = BrightnessContrastShader.BrightnessContrastShader;
exports.ColorCorrectionShader = ColorCorrectionShader.ColorCorrectionShader;
exports.ColorifyShader = ColorifyShader.ColorifyShader;
exports.ConvolutionShader = ConvolutionShader.ConvolutionShader;
exports.CopyShader = CopyShader.CopyShader;
exports.DOFMipMapShader = DOFMipMapShader.DOFMipMapShader;
exports.BlurShaderUtils = DepthLimitedBlurShader.BlurShaderUtils;
exports.DepthLimitedBlurShader = DepthLimitedBlurShader.DepthLimitedBlurShader;
exports.DigitalGlitch = DigitalGlitch.DigitalGlitch;
exports.DotScreenShader = DotScreenShader.DotScreenShader;
exports.FXAAShader = FXAAShader.FXAAShader;
exports.FilmShader = FilmShader.FilmShader;
exports.FocusShader = FocusShader.FocusShader;
exports.FreiChenShader = FreiChenShader.FreiChenShader;
exports.FresnelShader = FresnelShader.FresnelShader;
exports.GammaCorrectionShader = GammaCorrectionShader.GammaCorrectionShader;
exports.GodRaysCombineShader = GodRaysShader.GodRaysCombineShader;
exports.GodRaysDepthMaskShader = GodRaysShader.GodRaysDepthMaskShader;
exports.GodRaysFakeSunShader = GodRaysShader.GodRaysFakeSunShader;
exports.GodRaysGenerateShader = GodRaysShader.GodRaysGenerateShader;
exports.HalftoneShader = HalftoneShader.HalftoneShader;
exports.HorizontalBlurShader = HorizontalBlurShader.HorizontalBlurShader;
exports.HorizontalTiltShiftShader = HorizontalTiltShiftShader.HorizontalTiltShiftShader;
exports.HueSaturationShader = HueSaturationShader.HueSaturationShader;
exports.KaleidoShader = KaleidoShader.KaleidoShader;
exports.LuminosityHighPassShader = LuminosityHighPassShader.LuminosityHighPassShader;
exports.LuminosityShader = LuminosityShader.LuminosityShader;
exports.MirrorShader = MirrorShader.MirrorShader;
exports.NormalMapShader = NormalMapShader.NormalMapShader;
exports.ParallaxShader = ParallaxShader.ParallaxShader;
exports.PixelShader = PixelShader.PixelShader;
exports.RGBShiftShader = RGBShiftShader.RGBShiftShader;
exports.SAOShader = SAOShader.SAOShader;
exports.SMAABlendShader = SMAAShader.SMAABlendShader;
exports.SMAAEdgesShader = SMAAShader.SMAAEdgesShader;
exports.SMAAWeightsShader = SMAAShader.SMAAWeightsShader;
exports.SSAOBlurShader = SSAOShader.SSAOBlurShader;
exports.SSAODepthShader = SSAOShader.SSAODepthShader;
exports.SSAOShader = SSAOShader.SSAOShader;
exports.SSRBlurShader = SSRShader.SSRBlurShader;
exports.SSRDepthShader = SSRShader.SSRDepthShader;
exports.SSRShader = SSRShader.SSRShader;
exports.SepiaShader = SepiaShader.SepiaShader;
exports.SobelOperatorShader = SobelOperatorShader.SobelOperatorShader;
exports.SubsurfaceScatteringShader = SubsurfaceScatteringShader.SubsurfaceScatteringShader;
exports.TechnicolorShader = TechnicolorShader.TechnicolorShader;
exports.ToneMapShader = ToneMapShader.ToneMapShader;
exports.ToonShader1 = ToonShader.ToonShader1;
exports.ToonShader2 = ToonShader.ToonShader2;
exports.ToonShaderDotted = ToonShader.ToonShaderDotted;
exports.ToonShaderHatching = ToonShader.ToonShaderHatching;
exports.TriangleBlurShader = TriangleBlurShader.TriangleBlurShader;
exports.UnpackDepthRGBAShader = UnpackDepthRGBAShader.UnpackDepthRGBAShader;
exports.VerticalBlurShader = VerticalBlurShader.VerticalBlurShader;
exports.VerticalTiltShiftShader = VerticalTiltShiftShader.VerticalTiltShiftShader;
exports.VignetteShader = VignetteShader.VignetteShader;
exports.VolumeRenderShader1 = VolumeShader.VolumeRenderShader1;
exports.WaterRefractionShader = WaterRefractionShader.WaterRefractionShader;
exports.HTMLMesh = HTMLMesh.HTMLMesh;
exports.InteractiveGroup = InteractiveGroup.InteractiveGroup;
exports.SelectionHelper = SelectionHelper.SelectionHelper;
exports.SelectionBox = SelectionBox.SelectionBox;
exports.AmmoPhysics = AmmoPhysics.AmmoPhysics;
exports.ParallaxBarrierEffect = ParallaxBarrierEffect.ParallaxBarrierEffect;
exports.PeppersGhostEffect = PeppersGhostEffect.PeppersGhostEffect;
exports.OutlineEffect = OutlineEffect.OutlineEffect;
exports.AnaglyphEffect = AnaglyphEffect.AnaglyphEffect;
exports.AsciiEffect = AsciiEffect.AsciiEffect;
exports.StereoEffect = StereoEffect.StereoEffect;
exports.FBXLoader = FBXLoader.FBXLoader;
exports.Font = FontLoader.Font;
exports.FontLoader = FontLoader.FontLoader;
exports.TGALoader = TGALoader.TGALoader;
exports.LUTCubeLoader = LUTCubeLoader.LUTCubeLoader;
exports.NRRDLoader = NRRDLoader.NRRDLoader;
exports.STLLoader = STLLoader.STLLoader;
exports.MTLLoader = MTLLoader.MTLLoader;
exports.XLoader = XLoader.XLoader;
exports.BVHLoader = BVHLoader.BVHLoader;
exports.KMZLoader = KMZLoader.KMZLoader;
exports.VRMLoader = VRMLoader.VRMLoader;
exports.VRMLLoader = VRMLLoader.VRMLLoader;
exports.KTX2Loader = KTX2Loader.KTX2Loader;
exports.LottieLoader = LottieLoader.LottieLoader;
exports.TTFLoader = TTFLoader.TTFLoader;
exports.RGBELoader = RGBELoader.RGBELoader;
exports.AssimpLoader = AssimpLoader.AssimpLoader;
exports.ColladaLoader = ColladaLoader.ColladaLoader;
exports.MDDLoader = MDDLoader.MDDLoader;
exports.EXRLoader = EXRLoader.EXRLoader;
exports.ThreeMFLoader = _3MFLoader.ThreeMFLoader;
exports.XYZLoader = XYZLoader.XYZLoader;
exports.VTKLoader = VTKLoader.VTKLoader;
exports.LUT3dlLoader = LUT3dlLoader.LUT3dlLoader;
exports.DDSLoader = DDSLoader.DDSLoader;
exports.PVRLoader = PVRLoader.PVRLoader;
exports.GCodeLoader = GCodeLoader.GCodeLoader;
exports.BasisTextureLoader = BasisTextureLoader.BasisTextureLoader;
exports.TDSLoader = TDSLoader.TDSLoader;
exports.LDrawLoader = LDrawLoader.LDrawLoader;
exports.GLTFLoader = GLTFLoader.GLTFLoader;
exports.SVGLoader = SVGLoader.SVGLoader;
exports.Rhino3dmLoader = _3DMLoader.Rhino3dmLoader;
exports.OBJLoader = OBJLoader.OBJLoader;
exports.AMFLoader = AMFLoader.AMFLoader;
exports.MMDLoader = MMDLoader.MMDLoader;
exports.MD2Loader = MD2Loader.MD2Loader;
exports.KTXLoader = KTXLoader.KTXLoader;
exports.TiltLoader = TiltLoader.TiltLoader;
exports.DRACOLoader = DRACOLoader.DRACOLoader;
exports.HDRCubeTextureLoader = HDRCubeTextureLoader.HDRCubeTextureLoader;
exports.PDBLoader = PDBLoader.PDBLoader;
exports.PRWMLoader = PRWMLoader.PRWMLoader;
exports.RGBMLoader = RGBMLoader.RGBMLoader;
exports.VOXData3DTexture = VOXLoader.VOXData3DTexture;
exports.VOXLoader = VOXLoader.VOXLoader;
exports.VOXMesh = VOXLoader.VOXMesh;
exports.PCDLoader = PCDLoader.PCDLoader;
exports.LWOLoader = LWOLoader.LWOLoader;
exports.PLYLoader = PLYLoader.PLYLoader;
exports.LineSegmentsGeometry = LineSegmentsGeometry.LineSegmentsGeometry;
exports.LineGeometry = LineGeometry.LineGeometry;
exports.Wireframe = Wireframe.Wireframe;
exports.WireframeGeometry2 = WireframeGeometry2.WireframeGeometry2;
exports.Line2 = Line2.Line2;
exports.LineMaterial = LineMaterial.LineMaterial;
exports.LineSegments2 = LineSegments2.LineSegments2;
exports.LightProbeHelper = LightProbeHelper.LightProbeHelper;
exports.VertexTangentsHelper = VertexTangentsHelper.VertexTangentsHelper;
exports.PositionalAudioHelper = PositionalAudioHelper.PositionalAudioHelper;
exports.VertexNormalsHelper = VertexNormalsHelper.VertexNormalsHelper;
exports.RectAreaLightHelper = RectAreaLightHelper.RectAreaLightHelper;
exports.RectAreaLightUniformsLib = RectAreaLightUniformsLib.RectAreaLightUniformsLib;
exports.LightProbeGenerator = LightProbeGenerator.LightProbeGenerator;
exports.calcBSplineDerivatives = NURBSUtils.calcBSplineDerivatives;
exports.calcBSplinePoint = NURBSUtils.calcBSplinePoint;
exports.calcBasisFunctionDerivatives = NURBSUtils.calcBasisFunctionDerivatives;
exports.calcBasisFunctions = NURBSUtils.calcBasisFunctions;
exports.calcKoverI = NURBSUtils.calcKoverI;
exports.calcNURBSDerivatives = NURBSUtils.calcNURBSDerivatives;
exports.calcRationalCurveDerivatives = NURBSUtils.calcRationalCurveDerivatives;
exports.calcSurfacePoint = NURBSUtils.calcSurfacePoint;
exports.findSpan = NURBSUtils.findSpan;
exports.NURBSCurve = NURBSCurve.NURBSCurve;
exports.NURBSSurface = NURBSSurface.NURBSSurface;
exports.CinquefoilKnot = CurveExtras.CinquefoilKnot;
exports.DecoratedTorusKnot4a = CurveExtras.DecoratedTorusKnot4a;
exports.DecoratedTorusKnot4b = CurveExtras.DecoratedTorusKnot4b;
exports.DecoratedTorusKnot5a = CurveExtras.DecoratedTorusKnot5a;
exports.DecoratedTorusKnot5c = CurveExtras.DecoratedTorusKnot5c;
exports.FigureEightPolynomialKnot = CurveExtras.FigureEightPolynomialKnot;
exports.GrannyKnot = CurveExtras.GrannyKnot;
exports.HeartCurve = CurveExtras.HeartCurve;
exports.HelixCurve = CurveExtras.HelixCurve;
exports.KnotCurve = CurveExtras.KnotCurve;
exports.TorusKnot = CurveExtras.TorusKnot;
exports.TrefoilKnot = CurveExtras.TrefoilKnot;
exports.TrefoilPolynomialKnot = CurveExtras.TrefoilPolynomialKnot;
exports.VivianiCurve = CurveExtras.VivianiCurve;
exports.Face3 = Geometry.Face3;
exports.Geometry = Geometry.Geometry;
exports.MeshoptDecoder = MeshoptDecoder.MeshoptDecoder;
exports.MotionController = MotionControllers.MotionController;
exports.MotionControllerConstants = MotionControllers.MotionControllerConstants;
exports.fetchProfile = MotionControllers.fetchProfile;
exports.fetchProfilesList = MotionControllers.fetchProfilesList;
